lms/static/scripts/frontend_apps/components/StatusChip.tsx

```diff
@@ -0,0 +1,42 @@
+import classnames from 'classnames';
+import type { ComponentChildren } from 'preact';
+
+export type StatusChipProps = {
+  /**
+   * Determines the aspect of the status chip, being "success" the one
+   * representing the most successful status, and "fatal" the one representing
+   * a total failure.
+   */
+  variant:
+    | 'success'
+    | 'semi-success'
```

Comment:
I'm trying to come up with a better name than `semi-success`. Or perhaps use `success` for that one and think on something which is "above" success.

Reply:
The problem here is that you're trying to map a grading scale ranging from "best" to "worst" to operational status (success, failure). The former is a more nuanced and gradual scale. I can see a few options here:

1. Don't attempt to create a generic `StatusChip` yet, just have a grading-specific component
2. In the generic status chip use a letter or numeric scale as a score indicator. eg. `variant="grade-A"`. If you want to use something later that is functionally the same but indicates an operation status, you could have those mappings in addition (eg. "variant: "grade-A" | "grade-B" ... | "success"). Those variants might happen to share the same color

I would be inclined towards (1) because you don't yet know what behaviors you'll want exactly for a non-grade status chip.

Reply:
Yep, good point. I will go with 1.